### PR TITLE
Fix check errors for getaddrinfo and some other functions

### PR DESCRIPTION
**Pull request by ygj6**
Resolved some Windows/MinGW related errors  in configure.ac and CMakeLists.txt.
Fix https://github.com/libevent/libevent/issues/966

### CODE REVIEW

File: cmake/CheckSymbolsExist.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Check if each symbol in the symbol list exists,
+# and define PREFIX__HAVE_SYMNAME to 1 if yes.
+#
+# Global variables that it uses:
+# - CHECK_SYMBOL_EXTRA_HEADERS:LIST
+
+include(CheckSymbolExists)
+
+macro(CHECK_SYMBOLS_EXIST SYMLIST PREFIX)
+  list(JOIN CHECK_SYMBOL_EXTRA_HEADERS ";" CHECK_SYMBOL_EXTRA_HEADERS_JOINED)
```
Comment (azat) [inline cmake/CheckSymbolsExist.cmake]: `list(JOIN)` requires cmake 3.12, let's avoid using such recent cmake? (released in 2018-07)

P.S. fails travis-ci, although we should not care about this already https://travis-ci.org/github/libevent/libevent/jobs/663067267#L928

Comment (azat) [reply cmake/CheckSymbolsExist.cmake]: Actually AFAICS this line can be simply dropped and will just work (since list delimiter is `;` anyway)

Comment (ygj6) [reply cmake/CheckSymbolsExist.cmake]: Has been corrected

**Comment by ygj6** [comment]
> anything pending or this can be merged?

I think it can be merged.
There is another fail:
`FAIL ../test/regress_http.c:594: assert(test_ok == 2): -2 vs 2http/https_filter_basic: `
I will check at other times.

**Comment by azat** [comment]
You forget that git requires new line between commit subject and description I will rebase and merge after, thanks for taking care of this!

**Comment by ygj6** [comment]
I made a mistake here:
https://github.com/libevent/libevent/blob/1675a55620e6f0bbba5776f2df72cd48920421c2/CMakeLists.txt#L469
No matter which platform it runs on, it always uses `CHECK_SYMBOLS_EXIST`, which implies `CHECK_SYMBOL_EXISTS`. I have checked the results on Windows, MinGW and Ubuntu, and they are all ok. But I missed MacOS and it happened to be wrong here.

I want to change it to the following:
```
if (WIN32)
    CHECK_SYMBOLS_EXIST()
else()
    CHECK_FUNCTION_EXISTS()   # Consistent with previous code
endif()
```
should I `rebase/comimit` it or create a new PR?

**Comment by azat** [comment]
>should I rebase

You cannot rebase this, since it has been merged already.

> comimit

Yep, you can simply commit this.

> it or create a new PR?

You can just push such onelinear fixes I think.

>No matter which platform it runs on, it always uses CHECK_SYMBOLS_EXIST, which implies CHECK_SYMBOL_EXISTS. I have checked the results on Windows, MinGW and Ubuntu, and they are all ok. But I missed MacOS and it happened to be wrong here.

BTW why macos is different?

**Comment by ygj6** [comment]
The check results on MacOS after this PR are as follows:
```
-- Looking for getaddrinfo
-- Looking for getaddrinfo - not found
-- Looking for getnameinfo
-- Looking for getnameinfo - not found
-- Looking for getprotobynumber
-- Looking for getprotobynumber - not found
-- Looking for inet_ntop
-- Looking for inet_ntop - not found
-- Looking for inet_pton
-- Looking for inet_pton - not found
-- Looking for strtoll
-- Looking for strtoll - not found
-- Looking for timerclear
-- Looking for timerclear - not found
-- Looking for timercmp
-- Looking for timercmp - not found
-- Looking for timerisset
-- Looking for timerisset - not found
```
In fact, `getaddrinfo`, `getnameinfo` and `getprotobynumber` exist in `netdb.h`
 `inet_ntop` and `inet_pton` exist in `arpa/inet.h`
`timerclear`, `timercmp`, `timerisset`, `timeradd` and `timerisset` exist in `sys/time.h`

**Comment by ygj6** [comment]
Results before this PR are not entirely correct
```
......
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for getprotobynumber
-- Looking for getprotobynumber - found
-- Looking for getservbyname
-- Looking for getservbyname - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Looking for inet_pton
-- Looking for inet_pton - found
......
-- Looking for timeradd
-- Looking for timeradd - not found
-- Looking for timerclear
-- Looking for timerclear - not found
-- Looking for timercmp
-- Looking for timercmp - not found
-- Looking for timerfd_create
-- Looking for timerfd_create - not found
-- Looking for timerisset
-- Looking for timerisset - not found
-- Looking for putenv
-- Looking for putenv - found
-- Looking for setenv
-- Looking for setenv - found
......
```
`timerxxx` are not found

**Pull request merged**